src/main/java/mate.academy/Main.java

```diff
@@ -3,14 +3,15 @@
 import java.util.List;
 import mate.academy.lib.Injector;
 import mate.academy.model.Product;
-import mate.academy.service.ProductService;
+import mate.academy.service.impl.ProductServiceImpl;
 
 public class Main {
 
     public static void main(String[] args) {
         // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+        ProductServiceImpl productService = (ProductServiceImpl) injector
+                .getInstance(ProductServiceImpl.class);
         List<Product> products = productService.getAllFromFile("products.txt");
```

Comment:
Ensure that the method `getAllFromFile` is part of the `ProductService` interface, otherwise, you'll be breaking the abstraction by using an implementation-specific method.